actors/builtin/miner/deadline_state.go

```diff
@@ -1065,3 +1055,56 @@ func (dl *Deadline) RecordProvenSectors(
 		RetractedRecoveryPower: retractedRecoveryPowerTotal,
 	}, nil
 }
+
+func (dl *Deadline) RescheduleSectorExpirations(
+	store adt.Store, sectors Sectors, ssize abi.SectorSize, quant QuantSpec,
+	expiration abi.ChainEpoch, partitionSectors PartitionSectorMap,
+) error {
+	partitions, err := dl.PartitionsArray(store)
+	if err != nil {
+		return err
+	}
+
+	var rescheduledPartitions []uint64 // track partitions with moved expirations.
+	if err := partitionSectors.ForEach(func(partIdx uint64, sectorNos *abi.BitField) error {
+		var partition Partition
+		if found, err := partitions.Get(partIdx, &partition); err != nil {
+			return xc.ErrIllegalState.Wrapf("failed to load partition %d: %w", partIdx, err)
+		} else if !found {
+			// We failed to find the partition. That's ok, we'll skip it.
+			return nil
+		}
+
+		moved, err := partition.RescheduleExpirations(store, sectors, ssize, quant, expiration, sectorNos)
```

Comment:
Nit: I've been pushing the `ssize` and `qant` params to come last in every method they appear in.

Reply:
Ah, I've been doing the opposite (common args first, important args later). But either way is fine as long as we're consistent.